tests/func/test_add.py

```diff
@@ -513,13 +514,56 @@ def test(self):
 
 @pytest.mark.skipif(os.name != "nt", reason="Windows specific")
 def test_windows_should_add_when_cache_on_different_drive(
-    dvc_repo, repo_dir, temporary_windows_drive
+    tmp_dir, dvc, temporary_windows_drive
 ):
-    ret = main(["config", "cache.dir", temporary_windows_drive])
-    assert ret == 0
-
-    ret = main(["add", repo_dir.DATA])
-    assert ret == 0
+    tmp_dir.gen("file", "file content")
+
+    dvc.config.set("cache", "dir", temporary_windows_drive)
+    dvc = DvcRepo(dvc.root_dir)
+
+    stage, = dvc.add("file")
+    cache_path = stage.outs[0].cache_path
+
+    assert path_isin(cache_path, temporary_windows_drive)
+    assert os.path.isfile(cache_path)
+    filecmp.cmp("file", cache_path)
+
+
+@pytest.fixture
+def temporary_windows_drive(tmpdir_factory):
```

Comment:
-> `tmp_path_factory`

This aimed to replace `tmpdir_factory`, they have the migration of their own in pytest http://doc.pytest.org/en/latest/tmpdir.html#the-tmp-path-factory-fixture